packages/reactive-element/src/css-tag.ts

```diff
@@ -54,17 +53,15 @@ export class CSSResult {
     this.cssText = cssText;
   }
 
-  // Note, this is a getter so that it's lazy. In practice, this means
-  // stylesheets are not created until the first element instance is made.
+  // This is a getter so that it's lazy. In practice, this means stylesheets
+  // are not created until the first element instance is made.
   get styleSheet(): CSSStyleSheet | undefined {
-    // Note, if `supportsAdoptingStyleSheets` is true then we assume
-    // CSSStyleSheet is constructable.
-    let styleSheet = styleSheetCache.get(this.cssText);
-    if (supportsAdoptingStyleSheets && styleSheet === undefined) {
-      styleSheetCache.set(this.cssText, (styleSheet = new CSSStyleSheet()));
-      styleSheet.replaceSync(this.cssText);
+    // If `supportsAdoptingStyleSheets` is true then we assume CSSStyleSheet is
+    // constructable.
+    if (supportsAdoptingStyleSheets && this._styleSheet === undefined) {
+      (this._styleSheet = new CSSStyleSheet()).replaceSync(this.cssText);
```

Comment:
This is going to cause stylesheet duplication and css text re-parsing in some maybe still relatively common scenarios:

```ts
class BaseElement extends LitElement {
  static get styles() { return css`...`; }
}

class SubElementA extends BaseElement {
  static get styles() { return [BaseElement.styles, css`...`]; }
}
```

Reply:
It will however give a speedup on codebases that don't use a getter for `styles`, like all TS codebases, due to skipping a hash of a potentially long string.

I wonder if we can split the difference by caching based on the TemplateStringsArray, but only when the length is `1`, so there are no interpolations to worry about.

Reply:
This PR was based on https://github.com/lit/lit/pull/2978 and that change is made there, so let's discuss this change there?

Reply:
Oh yeah... I paused that one to consider what to do about getters.